{
  "comments": [
    {
      "unresolved": false,
      "key": {
        "uuid": "fbffe02c_f9c2ae79",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 1000153
      },
      "writtenOn": "2022-06-06T17:26:56Z",
      "side": 1,
      "message": "FWIW, I\u0027m really hoping there\u0027s a better design pattern we can find at some point for this kind of thing.  \"Do X when last reference to this is complete\" is a relatively common pattern, and I\u0027ve never been super happy about overloading the intent of the destructor for these use cases, but it seems to be the \"Best\" pattern we have so far (and for sure causes fewer bugs than making people manually reference count).",
      "revId": "27d5a076e6a00c5f1fcd2e5784f429937cab1a57",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "a9eeea6f_9bba8ade",
        "filename": "/PATCHSET_LEVEL",
        "patchSetId": 4
      },
      "lineNbr": 0,
      "author": {
        "id": 1000002
      },
      "writtenOn": "2022-06-06T19:20:40Z",
      "side": 1,
      "message": "\u003e for sure causes fewer bugs than making people manually reference count\n\nagreed!",
      "parentUuid": "fbffe02c_f9c2ae79",
      "revId": "27d5a076e6a00c5f1fcd2e5784f429937cab1a57",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "66bab65d_49207a0f",
        "filename": "src/main.cpp",
        "patchSetId": 4
      },
      "lineNbr": 122,
      "author": {
        "id": 1000153
      },
      "writtenOn": "2022-06-06T17:26:56Z",
      "side": 1,
      "message": "Should the debug timing also be in the try block?  I\u0027m not sure any of this can throw, but having the whole destructor in the try block seems more in line with the intent of this patchset.",
      "range": {
        "startLine": 108,
        "startChar": 0,
        "endLine": 122,
        "endChar": 6
      },
      "revId": "27d5a076e6a00c5f1fcd2e5784f429937cab1a57",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "b9ddd321_89edd08f",
        "filename": "src/main.cpp",
        "patchSetId": 4
      },
      "lineNbr": 122,
      "author": {
        "id": 1000002
      },
      "writtenOn": "2022-06-06T19:14:04Z",
      "side": 1,
      "message": "Makes sense to me - I went ahead and made this change.  Hopefully no-one sees the try/catch around the debug code (which...clang doesn\u0027t seem to think throws) and decides it\u0027s OK to call new code that generates exceptions without handling them explicitly.",
      "parentUuid": "66bab65d_49207a0f",
      "range": {
        "startLine": 108,
        "startChar": 0,
        "endLine": 122,
        "endChar": 6
      },
      "revId": "27d5a076e6a00c5f1fcd2e5784f429937cab1a57",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "75c0bb32_f4f907e6",
        "filename": "src/main.cpp",
        "patchSetId": 4
      },
      "lineNbr": 122,
      "author": {
        "id": 1000002
      },
      "writtenOn": "2022-06-06T19:20:57Z",
      "side": 1,
      "message": "done.",
      "parentUuid": "b9ddd321_89edd08f",
      "range": {
        "startLine": 108,
        "startChar": 0,
        "endLine": 122,
        "endChar": 6
      },
      "revId": "27d5a076e6a00c5f1fcd2e5784f429937cab1a57",
      "serverId": "adbd0a64-1f21-4d83-b585-671fe73cb6e4"
    }
  ]
}